File: Code/new_fitervari_backend/src/main/java/at/htl/model/WorkoutHistory.java
package at.htl.model;

import javax.json.bind.annotation.JsonbTransient;
import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "workout_history")
public class WorkoutHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime date;

    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    private Person person;

    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    @JoinColumn(name = "workout_id")
    @JsonbTransient
    private Workout workout;


    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "workout_history_id")
    private List<ExerciseHistory> exerciseHistories;

    //region GetterSetter
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    public Person getPerson() {
        return person;
    }

    public void setPerson(Person person) {
        this.person = person;
    }

    public Workout getWorkout() {
        return workout;
    }

    public void setWorkout(Workout workout) {
        this.workout = workout;
    }

    public List<ExerciseHistory> getExerciseHistories() {
        return exerciseHistories;
    }

    public void setExerciseHistories(List<ExerciseHistory> exerciseHistories) {
        this.exerciseHistories = exerciseHistories;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/WorkoutHistory.kt
package at.htl_leonding.model

import com.fasterxml.jackson.annotation.JsonIgnore
import io.quarkus.hibernate.orm.panache.PanacheEntity
import org.hibernate.jdbc.Work
import java.time.LocalDateTime
import javax.json.bind.annotation.JsonbTransient
import javax.persistence.*

@Entity
@Table(name = "workout_history")
class WorkoutHistory(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var date: LocalDateTime,
    @ManyToOne(cascade = arrayOf(CascadeType.MERGE))
    var customer: Customer,
    @ManyToOne(cascade = arrayOf(CascadeType.MERGE))
    @JoinColumn(name = "workout_id")
    @JsonbTransient
    var workout: Workout,
    @OneToMany(cascade = arrayOf(CascadeType.ALL))
    @JoinColumn(name = "workout_history_id")
    var exerciseHistories: MutableList<ExerciseHistory> = mutableListOf()
) {
    fun copyValues(other: WorkoutHistory) {
        this.date = other.date
        this.workout = other.workout
        this.customer = other.customer
        this.exerciseHistories = other.exerciseHistories
    }
}
